sharding/nodesCoordinator/hashValidatorShuffler.go

```diff
@@ -52,21 +52,21 @@ type randHashShuffler struct {
 	// when reinitialization of node in new shard is implemented
 	shuffleBetweenShards bool
 
-	adaptivity                               bool
-	nodesShard                               uint32
-	nodesMeta                                uint32
-	shardHysteresis                          uint32
-	metaHysteresis                           uint32
-	activeNodesConfig                        config.MaxNodesChangeConfig
-	availableNodesConfigs                    []config.MaxNodesChangeConfig
-	mutShufflerParams                        sync.RWMutex
-	validatorDistributor                     ValidatorsDistributor
-	flagBalanceWaitingLists                  atomic.Flag
-	enableEpochsHandler                      common.EnableEpochsHandler
-	stakingV4DistributeAuctionToWaitingEpoch uint32
-	flagStakingV4DistributeAuctionToWaiting  atomic.Flag
-	stakingV4EnableEpoch                     uint32
-	flagStakingV4                            atomic.Flag
+	adaptivity                bool
+	nodesShard                uint32
+	nodesMeta                 uint32
+	shardHysteresis           uint32
+	metaHysteresis            uint32
+	activeNodesConfig         config.MaxNodesChangeConfig
+	availableNodesConfigs     []config.MaxNodesChangeConfig
+	mutShufflerParams         sync.RWMutex
+	validatorDistributor      ValidatorsDistributor
+	flagBalanceWaitingLists   atomic.Flag
+	enableEpochsHandler       common.EnableEpochsHandler
+	stakingV4Step2EnableEpoch uint32
```

Comment:
I think this component can be simplified, perhaps in another PR(I should have thought about this when I integrated `enableEpochsHandler`)

as I see, enableEpochsHandler is only used to get `rhs.enableEpochsHandler.BalanceWaitingListsEnableEpoch()` which basically is a return from config

what if we pass to this component a custom `enableEpochsHandler`(not the main one from coreComponents), created with the real config, then on `UpdateShufflerConfig` we simply call `EpochConfirmed` on enableEpochsHandler:
```
epochSubscriberHandler, ok := rhs.enableEpochsHandler.(core.EpochSubscriberHandler)
	if !ok {
		log.Error("")
		return 
	}

	epochSubscriberHandler.EpochConfirmed(epoch, 0)
```

this way, all flags are updated + we don't have to manage here `flagStakingV4Step2, flagStakingV4Step3 and flagBalanceWaitingLists`, keeping this epoch-flag functionality inside `enableEpochsHandler`

Reply:
yup, great idea, we are already using this (check timeMachine.go file)

Reply:
added [task](https://multiversxlabs.atlassian.net/browse/MX-13842), on it

Reply:
#4977